### PR TITLE
Feature/process protected data usevoucher

### PR DESCRIPTION
**Pull request by luisccastillo**

### CODE REVIEW

File: packages/sdk/package-lock.json
Comment (PierreJeanjacquot) [inline packages/sdk/package-lock.json:1]: why so many changes here?

File: packages/sdk/src/lib/dataProtectorCore/processProtectedData.ts [hunk 1]
```diff
@@ -38,6 +39,7 @@ export const processProtectedData = async ({
   inputFiles,
   secrets,
   workerpool,
+  useVoucher,
```
Comment (PierreJeanjacquot) [inline packages/sdk/src/lib/dataProtectorCore/processProtectedData.ts:42]: I would set the default value for `useVoucher` here and revert the change made to `dataProtectorCore/IExecDataProtectorCore.ts`

```suggestion
  useVoucher = false,
```

File: packages/sdk/src/lib/dataProtectorCore/IExecDataProtectorCore.ts [hunk 2]
```diff
@@ -66,6 +66,7 @@ class IExecDataProtectorCore extends IExecDataProtectorModule {
     return processProtectedData({
       ...args,
       iexec: this.iexec,
+      useVoucher: args.useVoucher || false,
```
Comment (PierreJeanjacquot) [inline packages/sdk/src/lib/dataProtectorCore/IExecDataProtectorCore.ts:69]: the default value can be handled by `processProtectedData` implementation

```suggestion
```

File: packages/sdk/src/lib/dataProtectorCore/processProtectedData.ts [hunk 3]
```diff
@@ -154,10 +159,13 @@ export const processProtectedData = async ({
...
-    const { dealid, txHash } = await iexec.order.matchOrders({
-      requestorder,
-      ...underMaxPriceOrders,
-    });
+    const { dealid, txHash } = await iexec.order.matchOrders(
+      {
+        requestorder,
+        ...underMaxPriceOrders,
+      },
+      { preflightCheck: false, useVoucher: vUseVoucher }
```
Comment (PierreJeanjacquot) [inline packages/sdk/src/lib/dataProtectorCore/processProtectedData.ts:167]: wy adding `preflightCheck: false`?
disabling preflightCheck can create bad deals that produce always falling tasks.

```suggestion
      { useVoucher: vUseVoucher }
```

**Review by PierreJeanjacquot — CHANGES_REQUESTED** [review]
minor changes and topics to discuss

File: packages/sdk/package-lock.json
Comment (abbesBenayache) [reply packages/sdk/package-lock.json:1]: due to fixing iexec version and updating other packages after running npm install with Node.js 22

Comment (PierreJeanjacquot) [reply packages/sdk/package-lock.json:1]: ok, we should not do this in a feature PR

**Review by PierreJeanjacquot — APPROVED** [review]
